src/Integrations/Integrations/Swoole/SwooleIntegration.php

```diff
@@ -97,21 +97,14 @@ function (HookData $hook) use (&$rootSpan, $integration) {
                 $rootSpan->resource = $request->server['request_method'] . ' ' . $normalizedPath;
                 $rootSpan->meta[Tag::HTTP_METHOD] = $request->server['request_method'];
 
-                $serverProtocol = $request->server['server_protocol'] ?? 'HTTP/1.1';
-                $scheme = strpos($serverProtocol, 'HTTPS') !== false ? 'https://' : 'http://';
+                $scheme = $server->ssl ? 'https://' : 'http://';
                 $host = $headers['host'] ?? ($request->server['remote_addr'] . ':' . $request->server['server_port']);
                 $path = $request->server['request_uri'] ?? $request->server['path_info'] ?? '';
                 $query = isset($request->server['query_string']) ? '?' . $request->server['query_string'] : '';
                 $url = $scheme . $host . $path . $query;
                 $rootSpan->meta[Tag::HTTP_URL] = Normalizer::uriNormalizeincomingPath($url);
-            },
-            function (HookData $hook) use (&$rootSpan, $integration) {
-                if ($hook->exception) {
-                    $rootSpan->exception = $hook->exception;
-                }
 
-                \DDTrace\close_spans_until($rootSpan);
-                \DDTrace\close_span();
+                unset($rootSpan->meta['closure.declaration']);
```

Comment:
Why?

Reply:
Since we are tracing the [callback](https://github.com/laravel/octane/blob/6133c1cc13647903411d52d914501b681b65d1d2/bin/swoole-server#L109-L130), we would always have a redundant, non-really-useful tag with the location of this closure's declaration. In the case of Laravel Octane, for instance, we would always have this tag pointing to `swoole-server.php`, which is not really helpful.

Additionally, but to a lesser extent, this is inconsistent with other frameworks/libraries. If we are setting the location of the request callback declaration from Swoole, why aren't we setting the same for Laravel and `Kernel::handle`, for instance?

Reply:
Makes sense.